soc/xtensa/intel_adsp/tools/cavstwist.sh

```diff
@@ -65,12 +65,19 @@ set -e
 
 if [ -z "$CAVS_HOST" -o -z "$CAVS_KEY" -o -z "$CAVS_RIMAGE" ]; then
     echo "Missing cavstwist.sh configuration, must have:" 1>&2
-    echo "  export CAVS_HOST=ssh_host_name" 1>&2
+    echo "  export CAVS_HOST=ssh_host_name[:port]" 1>&2
     echo "  export CAVS_KEY=/path/to/signing_key.pem" 1>&2
     echo "  export CAVS_RIMAGE=/path/to/built/rimage/dir" 1>&2
     exit 1
 fi
 
+CAVS_PORT=${CAVS_HOST##[A-Za-z0-9@.]*:}
```

Comment:
This is a bit too restrictive and not IPv6-proof (fails with `[1:2::3:4]:678`). While there are Unix restrictions on hostnames, DNS actually supports unicode and what not and who knows what people could use in their username before @.

Strip the optional port number _first_ with `CAVS_HOST=${CAVS_HOST_PORT%:[0-9]*}`. If `CAVS_HOST != CAVS_HOST_PORT` then there is a port number, extract it with `CAVS_PORT=${CAVS_HOST_PORT#${CAVS_HOST}:}`

This still does not support IPv6 + port but it gets _much_ closer and it does not make any hostname assumption. 

PS: https://bugzilla.mindrot.org/show_bug.cgi?id=1602 - 13 years old ssh bug.